gems/pending/util/duplicate_blocker/dedup_handler.rb

```diff
@@ -63,8 +68,8 @@ def handle(method, *args)
       key = key_generator.call(method, *args)
       desc = descriptor.call(method, *args)
 
-      entry = update_history(key, desc)
-      tripped?(entry) ? on_blocking_call(desc) : method[*args]
+      entry = update_history(key.hash, desc)
```

Comment:
Yes we are expecting key generator to be provided by the user class in real use. Since we want to always use a hash code, we could demand key generator always returns a hash code though.